### PR TITLE
Script to collect Spam /Invalid PR across the GitHub Repo

### LINKED ISSUE #1025: Script to collect spam/invalid Pull Requests across GitHub Repositories
# Description

Create a Script to collect spam/invalid Pull Requests across GitHub Repositories. The Script should analyze the repositories stored in a CSV/JSON File and should capture the following data points: 

- Link of the PR 
- PR Author 
- Number of Files changed 
- Number of PR Comments 
- PR Reviewer(s) 

You can also add features to have the dataset collected in a CSV/JSON across all repositories or having separate CSV/JSON for each repository.

## Language

- [X] Go
- [X] Javascript
- [X] Python

### PR DESCRIPTION
**Pull request by sumitgsh**
# Description

The Script collects Spam /Invalid PR across the GitHub Repositories, all the details about the Repository will be collected from the repositories.json file and will be displayed in the command line
Fixes #1025

Replace `issue_no` with the issue number which is fixed in this PR

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sumitgsh** [comment]
@HarshCasper @vybhav72954 Could you please review and merge

File: Python/Git_Repo_Spam_Aggregator/repositories.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+[
+    {
+    "owner":"XXXXX",
+    "repoName": "XXXXXXXXXXX"
+    }
+]
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/repositories.json:7]: Add a new line at the End of the File

File: Python/Git_Repo_Spam_Aggregator/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- Link of the PR
+- PR Author
+- Number of files changed
+- Number of PR Comments
+- PR Reviewer(s)
+
+## Setup instructions
+
+- Install the required dependencies using `pip3 install -r requirements.txt`
+- Generate a `githubAPI` token and add the token to the `.env` file . To generate the token check this [tutorial](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/README.md:16]: Remove the space here `file  .`

File: Python/Git_Repo_Spam_Aggregator/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- Add Owner and RepoName to the repositories.json file .
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/README.md:17]: Add backticks on the `repositories.json`

Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/README.md:17]: Add a sample here on how the `repositories.json` can be developed

File: Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    # store Invalid PR content
+    PullRequestIDetails = []
+
+    # To store all the content of repositories
+    data = []
+
+    # Get the owner i.e username and repository name from the repositories.json file
+
+    with open("repositories.json") as file:
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py:52]: Add an exception handler to take into account the edge cases, when the `repositories.json` is empty or the repository does not exist.

File: Python/Git_Repo_Spam_Aggregator/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+PyGithub==1.55
+json
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/requirements.txt:2]: Add the apt JSON version: `2.0.9`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/Git_Repo_Spam_Aggregator/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Find Spam/Invalid Pull Request across GitHub-Repositries
+
+The script retrieves all the spam/invalid PRs from all the Github repositories provided in repositories.json and prints in the command line. The script captures the following data points
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/README.md:3]: - Replace PRs with Pull Requests
- Put back-tick around `repositories.json` 
- Put full stop at the end of the sentence

File: Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            data = json.load(file)
+
+            # Getting the individual repositories details
+            for repo in data:
+                repoName = g.get_repo(repo['owner']+"/"+repo['repoName'])
+                invalidPRdata = GetPullData(PullRequestIDetails, repoName)
+
+        except:
+            KeyError
+        print("The repositories.json file is empty")
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py:63]: Is this done properly? Can you validate it?

File: Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    
+    print("All the invalid/spam Pr details: ")
```
Comment (HarshCasper) [inline Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py:66]: Use Pull Request wherever possible. PR is ambiguous (even for some technical users)

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py (hunk 7 above)
Comment (sumitgsh) [reply Python/Git_Repo_Spam_Aggregator/get_spam_Invalid_pr.py:63]: Sorry, My bad this exception will be raised when there is no key in the dictionary .
-What I am thinking is just to use len() built-in function of python and validating whether the repository.json is empty or not..
-Second thing If the repository is not found then I will be using GithubException for that.
In my next Pr is will make the appropriate change

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**